cdap-api/src/main/java/co/cask/cdap/api/workflow/WorkflowContext.java

```diff
@@ -32,16 +30,16 @@
   long getLogicalStartTime();
 
   /**
-   * Returns a {@link Callable} that launches the associated program
-   * with the specified name when the {@link Callable#call() call} method is invoked. When the program completes,
-   * the {@link Callable} returns the {@link RuntimeContext} of the program or {@code null} if
+   * Returns a {@link Callable} that launches the associated program with the specified name when
+   * the {@link Callable#call() call} method is invoked. When the program completes,
+   * the {@link Callable} returns the {@link Map} of execution context of the program run or {@code null} if
    * no such context exists.
    * <p/>
    * <p> An Exception is thrown from the {@link Callable#call()} method if the program fails </p>
    *
    * @throws IllegalArgumentException if no program with the specified name is defined in the workflow
    */
-  Callable<RuntimeContext> getProgramRunner(String name);
+  Callable<Map<String, String>> getProgramRunner(String name);
```

Comment:
I think it's better to have an object to represent the execution context, even though inside that object is just a Map right now. It makes extension easier later.